File: DataStructure/String/08_ReverseVowelsofString.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        res = list(s)
        vowels = ['a','e','i','o','u']
        start = 0;
        end = len(s) - 1
        while start < end:
            while start < end and (s[start].lower() not in vowels):
                start += 1
            while start < end and (s[end].lower() not in vowels):
                end -= 1
            if(start != end):
                temp = res[start]
                res[start] = res[end]
                res[end] = temp
            start += 1
            end -= 1
        return ''.join(res)
        

File: DataStructure/String/08_ReverseVowelsofString.java
public class Solution {
    public String reverseVowels(String s) {
        int len = s.length();
        char[] chs = s.toCharArray();
        int i = 0, j = len - 1;
        while(i < j){
            if(!isVowels(chs[i])){
                i++;
                continue;
            }
            if(!isVowels(chs[j])){
                j--;
                continue;
            }
            char temp = chs[i];
            chs[i] = chs[j] ;
            chs[j] = temp;
            i++;
            j--;
            
        }
        return new String(chs);
    }
    public boolean isVowels(char c){
        if ('a' == c || 'e' == c || 'i' == c || 'o' == c || 'u' == c
                || 'A' == c || 'E' == c || 'I' == c || 'O' == c || 'U' == c) {
            return true;
        }
        return false;
    }
}